lib/ProgressPlugin.js

```diff
@@ -94,16 +110,16 @@ class ProgressPlugin {
 				}).apply(compiler);
 			});
 		} else {
-			let lastModulesCount = 0;
-			let moduleCount = 500;
+			let lastCount = 0;
+			let count = mode === "modules" ? modulesCount : 0;
 			let doneModules = 0;
 			const activeModules = [];
 
 			const update = module => {
 				handler(
-					0.1 + (doneModules / Math.max(lastModulesCount, moduleCount)) * 0.6,
-					"building modules",
-					`${doneModules}/${moduleCount} modules`,
+					0.1 + (doneModules / Math.max(lastCount, count)) * 0.6,
+					`building ${mode}`,
+					`${doneModules}/${count} ${mode}`,
```

Comment:
It would be much more useful if the handler took raw data (as an object or arguments in a particular order), rather than pre-made strings. E.g. in my implementation of a progress bar using the ProgressPlugin I have to resort to regexps/splits to re-parse the data out of the strings in order to be able to position them:

![progress-bar](https://user-images.githubusercontent.com/563469/47649441-99a62e80-db7d-11e8-88b9-6fe3c9eddbbc.gif)

If instead we provided a default handler that makes these strings above, we can get the same behavior, and still give greater flexibility to consumers, alternative handlers.

I would imagine handler being most flexible e.g. when called like this:

```js
handler({compilerIndex, currentStage, doneModules, activeModules, lastModulesCount, moduleCount, count, mode})
```

And some of those parameters would obviously be `undefined` in certain stages (which is fine). By stage I mean the hook currently being executed, or compilation.

Reply:
@niieani Seems like this is a breaking change, but in this PR I don't want to make any breaking changes
I think you may create your own PR for this feature to the [next](https://github.com/webpack/webpack/tree/next)-branch